sdks/python/apache_beam/io/vcfio.py

```diff
@@ -0,0 +1,378 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""A source for reading from VCF files (version 4.x).
+
+The 4.2 spec is available at https://samtools.github.io/hts-specs/VCFv4.2.pdf.
+"""
+
+from __future__ import absolute_import
+import collections
+
+from apache_beam.coders import coders
+from apache_beam.io.filesystem import CompressionTypes
+from apache_beam.io.iobase import Read
+from apache_beam.io import filebasedsource
+from apache_beam.io.textio import _TextSource as TextSource
```

Comment:
<!--thread_id:cc_144429489_t; commit:6587a5a34ad6aff2567df759a830067647adbaa5; resolved:0-->
<!--section:context-quote-->
> **jkff** wrote:
> I looked at the design doc and seems that @chamikaramj was fine with this. Okay then.

<!--section:body-->
We generally agreed to this so that we don't bloat the API of the ReadFromText() transform. Also, I think it might be useful to make BoundedSource objects available to other BoundedSource to be used through composition. Unfortunately Python doesn't have a notion of protected/package private access to differentiate APIs that are a part of public API from APIs that other modules can use but are not a part of public API.